_drafts/tracking-code-size.md

```diff
@@ -0,0 +1,598 @@
+---
+title: "Tracking Firmware Code Size"
+description:
+  "Tracking code size for embedded software projects can prevent surprises,
+  maintain productivity, and give insight to developers looking to retroactively
+  reduce a firmware binary size."
+tag: [fw-code-size]
+author: tyler
+image: /img/tracking-code-size/tracking-code-size.png
+---
+
+```
+arm-none-eabi-ld: firmware.elf section `.text' will not fit in region `ROM'
+arm-none-eabi-ld:: region `ROM' overflowed by 16 bytes
+make: error: arm-none-eabi-ld returned 1 exit status
+make: *** [firmware.elf] Error 1
+```
+
+This is the terminal output of my nightmares. It frequently means a halt in
+productivity and sends engineers scrambling to save 100 bytes wherever they can,
+all while trying to meet the deadlines and requirements set forth by the Product
+team.
+
+Fortunately, this message doesn't always have to come as a surprise. Where I've
+worked in the past, we had the tools and infrastructure in place to know when we
+would run out of code space, as well as where it all went. These tools were
+immensely helpful when our team at Pebble was trying to
+[cram an 800kB firmware into a 512kB ROM device](http://web.archive.org/web/20160308073714/https://blog.getpebble.com/2015/12/09/3ontintin/).
+
+Below is an example of a type of dashboard we had. When we approached 5kB
+remaining in our firmware image, we knew what to do.
+
+![](/img/tracking-code-size/running-out-code-space.png)
+
+Hopefully this post provides the tools you need to have a dashboard like this
+too!
+
+<!-- excerpt start -->
+
+Code size information should be surfaced and tracked throughout the development
```

Comment:
This feels a bit out of place. It makes sense as an excerpt, but not in the flow of the introduction.